accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,191 @@
+
+Users expect stability for projects which do not change, for example a project targeting an LTS release. It is much more difficult to ensure this stability when we need to significantly components which contribute complexity to that project.
+
+### Isn't this a lot of servicing if we find a bug?
+
+Although this seems like more changes than if behavior was in the compiler-distributed analyzers it is not.  If we cared enough about a bug to service an analyzer that runs on a stable .NET version we would also want to service the SDK for that stable .NET Version: including the same number of changes and branches.  In fact, this strategy *reduces* servicing in the case where a bug only exists in a single version of source generator.
+
+What this strategy does do is reduce risk.  We eliminate the channel where the latest SDK would contain changes that impacted users' projects targeting older .NET versions.  That is a good thing as it reduces the need to test those configurations and ensures we can garuntee the quality and stability of our stable .NET releases.
+
+### Is it bad to have multiple versions of a source generator possibly loaded by a repository?
+
+A repository may have projects targeting multiple frameworks.  These frameworks may contain different versions of the same source generator.  When building the repository this may require the same compiler instance to load multiple copies of a source generator.  The same compiler would not load multiple copies of source generators which ship in a non-framework specific location.  This should work and be supported based on source generator versioning.
+
+### Should all inbox source generators be tied to a shared framework?
+
+No, some source generators might be independent of a target framework. One example of such a source generators may not contain functionality but may instead just build some convenience API on top of existing public API that is unlikely to ever change.  These could be considered stable enough to place in a non-framework-specific location like the SDK's analyzer folder.  Source generators which choose to ship this way should be highly specified and will need to support backwards compatibility and test on all frameworks.
+
+### Should only source generators be shipped this way?  What about analyzers and code-fixes?
+
+Nothing is preventing analyzers and code fixes from shipping in the same manner.  If it is desirable for a such a component to ship with the framework rather than the compiler then that component may ship in this manner.  We should only do this if we have a good reason, stable analyzers that are lanaguage extensions or tied to stable public API should ship non-framework-specific with the SDK as they do today.  Analyzers that need to validate source generator usage, or analyzers that are closely tied to runtime functionality may consider shipping with the framework.
```

Comment:
```suggestion
Nothing is preventing analyzers and code fixes from shipping in the same manner.  If it is desirable for a such a component to ship with the framework rather than the compiler then that component may ship in this manner.  We should only do this if we have a good reason; stable analyzers that are language extensions or tied to stable public API should ship non-framework-specific with the SDK as they do today.  Analyzers that need to validate source generator usage, or analyzers that are closely tied to runtime functionality may consider shipping with the framework.
```